@babel/parser throw an error when parsing object shorthand property async get () after updating to version 7.11.1

`index.js`
```js
export default {
  async get () {}
};
```
`package.json`
```json
{
  "name": "babel-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {},
  "devDependencies": {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.10.3"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "babel": "babel index.js"
  },
  "author": "",
  "license": "ISC"
}
```

when running `npm run babel`, I got an error below
```bash
{ SyntaxError: /Users/icefox/Desktop/babel-test/index.js: Unexpected token (2:11)

  1 | export default {
> 2 |   async get () {}
    |             ^
  3 | };
    at Parser._raise (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:763:17)
    at Parser.raiseWithData (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:756:17)
    at Parser.raise (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:750:17)
    at Parser.unexpected (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:8930:16)
    at Parser.parseIdentifierName (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:11017:18)
    at Parser.parseIdentifier (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:10994:23)
    at Parser.parseMaybePrivateName (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:10311:19)
    at Parser.parsePropertyName (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:10795:155)
    at Parser.parsePropertyDefinition (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:10708:14)
    at Parser.parseObjectLike (/Users/icefox/Desktop/babel-test/node_modules/_@babel_parser@7.11.1@@babel/parser/lib/index.js:10612:25)
  loc: Position { line: 2, column: 11 },
  pos: 28,
  code: 'BABEL_PARSE_ERROR' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! babel-test@1.0.0 babel: `babel index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the babel-test@1.0.0 babel script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/icefox/.tnpm/_logs/2020-08-05T05_53_09_257Z-debug.log
[tnpm] subprocess:/Users/icefox/.nvm/versions/node/v10.21.0/lib/node_modules/tnpm/lib/npm_child_process.js exit code: 1, signal: null
[tnpm] mainprocess exit code: 1
```

It works fine with `async` with property name except `get`, and older version of `@babel/parser`, and in latest chrome and node v8, it turns out a valid syntax

--- a/packages/babel-parser/src/parser/expression.js
+++ b/packages/babel-parser/src/parser/expression.js
@@ -1754,7 +1754,7 @@ export default class ExpressionParser extends LValParser {
       }
       // get PropertyName[?Yield, ?Await] () { FunctionBody[~Yield, ~Await] }
       // set PropertyName[?Yield, ?Await] ( PropertySetParameterList ) { FunctionBody[~Yield, ~Await] }
-      if (prop.key.name === "get" || prop.key.name === "set") {
+      else if (prop.key.name === "get" || prop.key.name === "set") {
         isAccessor = true;
         isGenerator = this.eat(tt.star); // tt.star is allowed in `maybeAsyncOrAccessorProp`, we will throw in `parseObjectMethod` later
         prop.kind = prop.key.name;

--- a/packages/babel-parser/test/fixtures/es2017/async-functions/async-shorthand-method/input.js
+++ b/packages/babel-parser/test/fixtures/es2017/async-functions/async-shorthand-method/input.js
@@ -1,0 +1,4 @@
+({
+  async get() {},
+  async set() {}
+})

--- a/packages/babel-parser/test/fixtures/es2017/async-functions/async-shorthand-method/output.json
+++ b/packages/babel-parser/test/fixtures/es2017/async-functions/async-shorthand-method/output.json
@@ -1,0 +1,71 @@
+{
+  "type": "File",
+  "start":0,"end":40,"loc":{"start":{"line":1,"column":0},"end":{"line":4,"column":2}},
+  "program": {
+    "type": "Program",
+    "start":0,"end":40,"loc":{"start":{"line":1,"column":0},"end":{"line":4,"column":2}},
+    "sourceType": "script",
+    "interpreter": null,
+    "body": [
+      {
+        "type": "ExpressionStatement",
+        "start":0,"end":40,"loc":{"start":{"line":1,"column":0},"end":{"line":4,"column":2}},
+        "expression": {
+          "type": "ObjectExpression",
+          "start":1,"end":39,"loc":{"start":{"line":1,"column":1},"end":{"line":4,"column":1}},
+          "properties": [
+            {
+              "type": "ObjectMethod",
+              "start":5,"end":19,"loc":{"start":{"line":2,"column":2},"end":{"line":2,"column":16}},
+              "method": true,
+              "key": {
+                "type": "Identifier",
+                "start":11,"end":14,"loc":{"start":{"line":2,"column":8},"end":{"line":2,"column":11},"identifierName":"get"},
+                "name": "get"
+              },
+              "computed": false,
+              "kind": "method",
+              "id": null,
+              "generator": false,
+              "async": true,
+              "params": [],
+              "body": {
+                "type": "BlockStatement",
+                "start":17,"end":19,"loc":{"start":{"line":2,"column":14},"end":{"line":2,"column":16}},
+                "body": [],
+                "directives": []
+              }
+            },
+            {
+              "type": "ObjectMethod",
+              "start":23,"end":37,"loc":{"start":{"line":3,"column":2},"end":{"line":3,"column":16}},
+              "method": true,
+              "key": {
+                "type": "Identifier",
+                "start":29,"end":32,"loc":{"start":{"line":3,"column":8},"end":{"line":3,"column":11},"identifierName":"set"},
+                "name": "set"
+              },
+              "computed": false,
+              "kind": "method",
+              "id": null,
+              "generator": false,
+              "async": true,
+              "params": [],
+              "body": {
+                "type": "BlockStatement",
+                "start":35,"end":37,"loc":{"start":{"line":3,"column":14},"end":{"line":3,"column":16}},
+                "body": [],
+                "directives": []
+              }
+            }
+          ],
+          "extra": {
+            "parenthesized": true,
+            "parenStart": 0
+          }
+        }
+      }
+    ],
+    "directives": []
+  }
+}

